### DISCUSSION TITLE
Chapter(4):Why do we need to defer calling next() in task callback?

### DISCUSSION THREAD

**Discussion by thetnaingtn** [PacktPublishing/Node.js-Design-Patterns-Third-Edition#33] [Q&A]
In Chapter(4) Limiting concurrency
In the next method of TaskQueue class, in the callback which is passed to task() function, why next() function call is deferred to the next circle. Can somebody please tell me why?
`task(()=>{
    this.running--
    process.nextTick(this.next.bind(this)) //why do we need to defer calling next function ?
})`
Thanks in advance.

**Comment by JacobNewton007** [comment]
I think the next() is deferred because its a callback to get called  when we have still task in the queue
